wooz1ewu/ITechArt_Front-End
# React develop

# Overview  
> 1. Create project using create-react-app.  
> 2. Implement Note-info displaying.  
> 3. Implement full info displaying of active(selected) note.  
> 4. Add prop-types checkings.  
> 5. Implement storing of active note via useState hook.  
> 6. Use material-ui.
> 7. Implement project structure.  

# Changes  
> 1. Project structure was created using BEM.
> 2. MyNotes page's side panel and active-note display panel implemented.  
> 3. NoteList, NoteItem, DisplayedNote components were implemented so to display all the necessary info.
> 4. Styles were added and implemented using constant-objects.
> 5. Active note and note list are stored using useState hook.
> 6. Close button implemented with material-ui button.  
> 7. Prop-types checkings implemented using prop-types module.  
> 8. App was deployed on githib pages.

# How to test
> 1. Load [page](https://wooz1ewu.github.io/ITechArt_Front-End/), check if notes are displayed.   
> 1.1. Check if "Choose a note to display" message shown while note is not selected.
> 2. Set active note by clicking on it.  
> 2.1. Check if active note is illuminated.  
> 2.2. Check if active note information is displayed on display-panel.
